opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -105,29 +111,14 @@ public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
 
         // init if the tracer is null
-        if (tracer == null) {
-            ServletContext servletContext = filterConfig.getServletContext();
-
-            Object contextAttribute = servletContext.getAttribute(TRACER);
-            if (contextAttribute == null) {
-                log.severe("Tracer was not found in `ServletContext.getAttribute(TRACER)`, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            if (!(contextAttribute instanceof Tracer)) {
-                log.severe("Tracer from `ServletContext.getAttribute(TRACER)`, is not an instance of " +
-                        "io.opentracing.Tracer, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            this.tracer = (Tracer)contextAttribute;
-
-            this.spanDecorators = (List<SpanDecorator>)servletContext.getAttribute(SPAN_DECORATORS);
-            if (this.spanDecorators == null) {
-                this.spanDecorators = Arrays.asList(SpanDecorator.STANDARD_TAGS);
+        if (spanDecorators == null) {
+            spanDecorators = decoratorSPI();
+            if (spanDecorators.isEmpty()) {
+                this.spanDecorators = Arrays.<ServletFilterSpanDecorator>asList(new ServletFilterSpanDecorator.StandardTags());
             }
-            this.spanDecorators.removeAll(Collections.singleton(null));
         }
+
+        log.info("Using servlet filter span decorators:" + spanDecorators);
```

Comment:
Is your `spanDecorator` printing a list? Also, the way it's done, it's building a string from all span decorators, no matter if `info` is enabled for this logger. So, it might be better to use something like this: `log.info(() -> String.format("Using servlet filter span decorators: %s", spanDecorators));` . This way, the lambda is evaluated *only* if info is enabled for the logger.

Or wrap it within a `log.isInfoEnabled()`, if you can't use lambdas.